src/components/recordedit/select-all-row.tsx

```diff
@@ -0,0 +1,405 @@
+        logRecordeditClientAction,
+    } = useRecordedit();
+    const {
+        watch,
+        reset,
+        getValues,
+        formState: { errors },
+    } = useFormContext();
+    const ref = useRef(null);
+    const [isEmpty, setIsEmpty] = useState(true);
+
+    // This is to set select all checkbox state
+    const [selectAll, setSelectAll] = useState(false);
+
+    // This is to toggle the visibility of tooltip on checkbox container. The tooltip should disappear once checkbox container is clicked.
+    const [showTooltip, setShowTooltip] = useState(false);
+
+    // This is to set if there is a change in the width of select all area on resize and zoom
+    const [selectAllWidthChanged, setSelectAllWidthChaged] = useState(0);
+
+    // This useeffect is to set the first form as active on load
+    useEffect(() => {
+        if (activeForms?.length === 0) {
+            setActiveForm([forms[0]]);
+        }
+    }, []);
+
+    /* This useffect is to set the indeterminate checkbox when forms are individually selected
+     *  and selected forms length is less than total forms length
+     */
+    useEffect(() => {
+        if (ref && ref.current && activeForms) {
+            (ref.current as HTMLInputElement).indeterminate =
+                activeForms?.length > 0
+                    ? activeForms?.length < forms.length
+                        ? true
+                        : false
+                    : false;
+            setSelectAll(activeForms?.length === forms.length);
+        }
+    }, [activeForms]);
+
+    // useEffect to call update only when there is a change in the width of select-all-row to update textarea width
+    useEffect(() => {
+        updateTextareaWidth();
+    }, [selectAllWidthChanged]);
+
+    /* useEffect to have a resize sensor to change the direction of upper-row when width of container is less than 400.
+       We chose 400, based on manual testing and thats when the buttons and checbox container starts overlapping.
+       Since we want to reduce the width of text area till min-width 250px we are moving the buttons and checbox container
+       to two rows with button container on top.
+    */
+    useEffect(() => {
+
+        const inputDiv = document.querySelector('.select-all-input') as HTMLElement;
+
+        const mainResizeSensor = new ResizeSensor(inputDiv,
+            () => {
+                const upperRow = document.querySelector('.select-upper-row') as HTMLElement;
+                const upperRowWidth = upperRow?.getBoundingClientRect().width;
+                if (upperRow && upperRowWidth < 400) {
+                    upperRow.style.flexDirection = 'column-reverse'
+                } else {
+                    upperRow.style.flexDirection = 'row'
+                }
+            }
+        );
+
+        return () => {
+            mainResizeSensor.detach();
+        };
+    }, []);
+
+    // useEffect to have a resize sensor to width of textarea to the the parent container width.
```

Comment:
Your comment doesn't make sense to me. Can you please rewrite this and explain it better? Like I mentioned you should explain but what this function is doing, and why you're doing this.